evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,403 @@
+                newIndex = reviewed.length;
+            } else {
+                newIndex = 0;
+            }
+
+            items = reviewed.concat(unreviewed);
+            let direction = newIndex > index? "right" : "left";
+            index = newIndex;
+            active = items[index];
+            slideLayer(2, direction, active);
+        }
+
+        function press(action, toggle) {
+            if(!active  || action === "make_private" && !active.data("contribution")) {
+                return;
+            }
+
+            if(toggle && active.attr("data-review") === action) {
+                action = "unreview";
+            }
+
+            let parameters = {"id": active.data("id"), "action": action, "course_id": {{ course.id }}};
+            $.ajax({
+                type: "POST",
+                url: "{% url 'staff:course_comments_update_publish' %}",
+                data: parameters,
+                error: function(){ window.alert("{% trans 'The server is not responding.' %}"); }
+            });
+
+            if(action === "unreview") {
+                active.removeAttr("data-review");
+                updateButtons();
+                return;
+            }
+
+            active.attr("data-review", action);
+            updateButtons();
+
+            slide("right");
+            slider.find("[data-action=" + action + "]").focus();
+        }
+
+        function updateState() {
+            slider.find(".slider-side-left").toggleClass("shown", index > 0);
+            let reviewed = items.slice(0, index).filter(item => item.is("[data-review]")).length;
+            slider.find("[data-counter=reviewed-left]").text(reviewed);
+            slider.find("[data-counter=unreviewed-left]").text(index - reviewed);
+
+            slider.find(".slider-side-right").toggleClass("shown", active !== null);
+            reviewed = items.slice(index + 1).filter(item => item.is("[data-review]")).length;
+            slider.find("[data-counter=reviewed-right]").text(reviewed);
+            let unreviewed = items.length - index - reviewed - 1;
+            slider.find("[data-counter=unreviewed-right]").text(Math.max(unreviewed, 0));
+
+            if(!active) {
+                let reviewed = items.every(item => item.is("[data-review]"));
+                let alert = slider.find(".alert");
+                alert.toggleClass("alert-warning", !reviewed);
+                alert.toggleClass("alert-success", reviewed && items.length !== 0);
+                alert.toggleClass("alert-secondary", reviewed && items.length === 0);
+                alert.find("[data-content=unreviewed]").toggleClass("d-none", reviewed);
+                alert.find("[data-content=reviewed]").toggleClass("d-none", !reviewed);
+                slider.find("[data-action=repeat-unreviewed]").toggleClass("d-none", reviewed);
+                slider.find("[data-action=repeat-all]").toggleClass("d-none", items.length === 0);
+            }
+        }
```

Comment:
do you actually *want* to update all of this on every call to `handle`? please check whether there's a reasonable way to split this up and call the parts only when needed.